packages/svelte/src/internal/client/validate.js

```diff
@@ -67,6 +76,30 @@ export function validate_each_keys(collection, key_fn) {
 	}
 }
 
+/**
+ * @param {any} value
+ */
+function is_possiblly_reactive_object(value) {
+	if (STATE_SYMBOL in value) {
+		return true;
+	}
+	const prototype = get_prototype_of(value);
+	if (prototype === array_prototype || prototype === object_prototype) {
+		const descriptors = get_descriptors(value);
+		for (let key in descriptors) {
+			var prop_value = value[key];
+			if (
+				descriptors[key].set ||
+				(typeof prop_value === 'object' &&
+					prop_value !== null &&
+					is_possiblly_reactive_object(prop_value))
+			)
+				return true;
+		}
+	}
+	return false;
```

Comment:
should we also return true here if we detect that this is a class? because we're not going into those, and they could contain setters

Reply:
Good point